### PR TITLE
msubworkflow2

### PR DESCRIPTION
**Pull request by sysbiocoder**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR> -stub`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by Lucpen — COMMENTED** [review]
Almost there now :D

File: subworkflows/local/align_and_call_MT.nf [hunk 1]
```diff
@@ -31,15 +33,16 @@ workflow ALIGN_AND_CALL_MT {
         ch_fastq_ubam  = ch_mt_bam.join(ubam, by: [0])
 
         // Merges bam files
-        GATK4_MERGEBAMALIGNMENT_MT ( ch_fastq_ubam, fasta, dict )
+        GATK4_MERGEBAMALIGNMENT_MT (ch_fastq_ubam, fasta, dict )
         ch_versions = ch_versions.mix(GATK4_MERGEBAMALIGNMENT_MT.out.versions.first())
 
+       
         // Add read group to merged bam file
```
Comment (Lucpen) [inline subworkflows/local/align_and_call_MT.nf:40]:
```suggestion
        GATK4_MERGEBAMALIGNMENT_MT (ch_fastq_ubam, fasta, dict )
        ch_versions = ch_versions.mix(GATK4_MERGEBAMALIGNMENT_MT.out.versions.first())
       
        // Add read group to merged bam file
```

File: subworkflows/local/analyse_MT.nf [hunk 2]
```diff
@@ -35,12 +38,47 @@ workflow ANALYSE_MT {
...
+        // STEP 2.2: MT ALLIGNMENT SHIFT AND VARIANT CALLING
+        ch_intervals_mt_shift = Channel.fromPath(params.intervals_mt_shift)
+        PREPARE_GENOME_MT("bwamem2",[],[],fasta_shift ,[],[],[]).set { ch_genome }
+        ch_versions = ch_versions.mix(ch_genome.versions)
+        ch_dict_shift = ch_genome.sequence_dict
+        ch_fai_shift = ch_genome.fai
+        ch_index_shift =ch_genome.aligner_index
+
+
+        ALIGN_AND_CALL_MT_SHIFT (
```
Comment (Lucpen) [inline subworkflows/local/analyse_MT.nf:51]:
```suggestion
        ch_index_shift =ch_genome.aligner_index

        ALIGN_AND_CALL_MT_SHIFT (
```

File: subworkflows/local/prepare_genome.nf [hunk 3]
```diff
@@ -29,7 +29,7 @@ workflow PREPARE_GENOME {
             ch_aligner_index  = !bwamem2_index ? BWAMEM2_INDEX.out.index : [[],file(bwamem2_index)]
             ch_versions = ch_versions.mix(BWAMEM2_INDEX.out.versions)
         } else if (aligner == "sentieon") {
-            SENTIEON_BWAINDEX ( [[], ch_fasta] )
+            SENTIEON_BWAINDEX ([ [], ch_fasta] )
```
Comment (Lucpen) [inline subworkflows/local/prepare_genome.nf:32]:
```suggestion
            SENTIEON_BWAINDEX ( [[], ch_fasta] )
```

**Review by Lucpen — COMMENTED** [review]
Just a couple of tiny things

File: conf/modules.config [hunk 4]
```diff
@@ -700,6 +700,70 @@ process {
...
+        publishDir = [
+            path: { "${params.outdir}/references" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename },
+            enabled: params.save_reference,
+            pattern: "*{0123,amb,ann,bwt.2bit.64,bwt.8bit.32,pac}"
+        ]
+    }
+
+
```
Comment (ramprasadn) [inline conf/modules.config:718]: I think the general standard is to leave just one line between two modules here to help with readability. So my suggestion would be to delete the extra empty lines here.

File: conf/modules.config [hunk 5]
```diff
@@ -700,6 +700,70 @@ process {
...
+            pattern: "*fai"
+        ]
+    }
+
+    withName: '.*ANALYSE_MT:PREPARE_GENOME_MT:PREPARE_GENOME:GET_CHROM_SIZES' {
+        publishDir = [
+            enabled: false,
+        ]
+    }
+    withName: '.*ANALYSE_MT:ALIGN_AND_CALL_MT_SHIFT:BWAMEM2_MEM_MT' {
```
Comment (ramprasadn) [inline conf/modules.config:735]:
```suggestion

        withName: '.*ANALYSE_MT:ALIGN_AND_CALL_MT_SHIFT:BWAMEM2_MEM_MT' {
```

File: conf/modules.config [hunk 6]
```diff
@@ -700,6 +700,70 @@ process {
...
+        ext.prefix = { "${meta.id}_sorted" }
+    }
+
+
+    withName: '.*ANALYSE_MT:ALIGN_AND_CALL_MT_SHIFT:GATK4_MUTECT2_MT' {
+        ext.args = '--mitochondria-mode TRUE'
+        }
+    withName: '.*ANALYSE_MT:PICARD_LIFTOVERVCF' {
+        ext.args = ''
+        }
```
Comment (ramprasadn) [inline conf/modules.config:763]: I guess you can remove this if you are not changing the defaults for that module

**Pull request closed without merging**